src/gtk/view_dialog.c

```diff
@@ -44,8 +44,20 @@ do_view_or_edit_file (gftp_window_data * fromwdata, int is_view)
   if (S_ISDIR (curfle->st_mode))
     {
       if (is_view)
-        ftp_log (gftp_logging_error, NULL,
-                 _("View: %s is a directory. Cannot view it.\n"), curfle->file);
+      {
+        if (strcmp (gftp_protocols[fromwdata->request->protonum].name, "Local") == 0)
+        {
+          char cmd[2048]={'\0'};
+          sprintf(cmd,"xdg-open '%s'",curfle->file);
```

Comment:
Use `g_snprintf` instead of `sprintf` and you can remove the NUL byte for cmd above.

Reply:
Actually, you don't want to do that. You need to use something like `execve` instead. fork_process() in src/gtk/view_dialog.c has an example. Really that function should be split up and parts moved into lib/.

Reply:
I also suggest checking to see if glib has some functionality to do this since that may make it easier and more secure.